editor/src/clj/editor/grid.clj

```diff
@@ -13,192 +13,385 @@
+              grid-size-large
+              (conj (snap-out-to-grid aabb grid-size-large)))
+     :plane plane}))
+
+(g/defnk produce-merged-options
+  [prefs camera options]
+  (cond-> (if prefs (prefs/get prefs grid-prefs-path) {})
+    :always
+    (assoc :auto-scale true)
+
+    options
+    (merge options)
+
+    (c/mode-2d? camera)
+    (assoc :active-plane :z)))
 
 (g/defnode Grid
+  (property prefs g/Any)
+
   (input camera Camera)
-  (property grid-color types/Color)
-  (property auto-grid  g/Bool)
-  (property fixed-grid-size g/Int
-            (default 0))
-  (output grids      g/Any :cached update-grids)
-  (output renderable pass/RenderData :cached grid-renderable))
 
+  (output options g/Any (g/constantly nil))
+  (output merged-options g/Any produce-merged-options)
+  (output grids g/Any :cached produce-grids)
+  (output renderable pass/RenderData :cached produce-renderable))
+
+(defn- invalidate-grids! [app-view]
+  (let [scene-view-id (g/node-value app-view :active-view)
+        grid-id (g/node-value scene-view-id :grid)]
+    (g/transact [(g/invalidate-output grid-id :grids)])))
+
+(defmulti settings-row (fn [_app-view _prefs option] option))
+
+(defmethod settings-row :opacity
+  [app-view prefs option]
+  (let [prefs-path (conj grid-prefs-path option)
+        value (prefs/get prefs prefs-path)
+        slider (Slider. 0.0 1.0 value)
+        label (doto (Label. "Opacity")
+                (.setPrefWidth 70))]
+    (ui/observe
+     (.valueProperty slider)
+     (fn [_observable _old-val new-val]
+       (let [val (math/round-with-precision new-val 0.01)]
+         (prefs/set! prefs prefs-path val)
+         (invalidate-grids! app-view))))
+    [label slider]))
+
+(defn plane-toggle-button
+  [prefs plane-group prefs-path plane]
+  (let [active-plane (prefs/get prefs prefs-path)]
+    (doto (ToggleButton. (string/upper-case (name plane)))
+      (.setToggleGroup plane-group)
+      (.setSelected (= plane active-plane))
+      (ui/add-style! "plane-toggle"))))
+
+(defmethod settings-row :active-plane
+  [app-view prefs option]
+  (let [prefs-path (conj grid-prefs-path option)
+        plane-group (ToggleGroup.)
+        buttons (mapv (partial plane-toggle-button prefs plane-group prefs-path) axes)
+        label (doto (Label. "Plane")
+                (HBox/setHgrow Priority/ALWAYS)
+                (.setPrefWidth 70))]
+    (ui/observe (.selectedToggleProperty plane-group)
+                (fn [_ ^ToggleButton old-value ^ToggleButton new-value]
+                  (if new-value
+                    (do (let [active-plane (-> (.getText new-value) string/lower-case keyword)]
+                          (prefs/set! prefs prefs-path active-plane))
+                        (invalidate-grids! app-view))
+                    (.setSelected old-value true))))
+    (concat [label] buttons)))
+
+(defmethod settings-row :color
+  [app-view prefs option]
+  (let [prefs-path (conj grid-prefs-path option)
+        text-field (TextField.)
```

Comment:
I wanted to avoid showing a popover within a popover, so I decided to display a simple text field. I am not thrilled about this.